ember-client/shared/src/main/scala/org/http4s/ember/client/internal/WebSocketHelpers.scala

```diff
@@ -37,6 +43,55 @@ private[client] object WebSocketHelpers {
   private[internal] val webSocketProtocol = Protocol(ci"websocket", None)
   private[internal] val connectionUpgrade = Connection(NonEmptyList.of(upgradeCi))
   private[internal] val upgradeWebSocket = Upgrade(webSocketProtocol)
+  private[internal] val exampleSecWebSocketKey = "dGhlIHNhbXBsZSBub25jZQ=="
+
+  // private[this] val clientTranscoder = new FrameTranscoder(true)
+
+  private[internal] def createWebSocketKey[F[_]]: Key[Socket[F]] =
+    Key.newKey[SyncIO, Socket[F]].unsafeRunSync()
+
+  def getSocket[F[_]](client: Client[F], request: Request[F])(implicit
+      F: MonadCancel[F, Throwable]
+  ): Resource[F, Resource[F, Option[Socket[F]]]] = {
+    val webSocketKey = createWebSocketKey[F]
+    client
+      .run(request)
+      .map { res =>
+        Resource.eval(validateServerHandshake(res, exampleSecWebSocketKey)).map { isValid =>
+          isValid match {
+            case Left(_) => None
```

Comment:
If the body of a `map`, or `flatMap`, or other higher-order-function, has the form of a pattern-matching, it is possible to omit the variable and just write the cases.

```suggestion
        Resource.eval(validateServerHandshake(res, exampleSecWebSocketKey)).map { 
          case Left(_) => None
```